src/postgres/abstract/maps.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as FlexibleModels from "@azure/arm-postgresql-flexible/esm/models";
+import * as SingleModels from "@azure/arm-postgresql/esm/models";
+import { AbstractServerCreate, PostgresAbstractDatabase, PostgresAbstractServer, PostgresServerType } from "./models";
+
+
+
+export function singleAsAbstractServer(server: SingleModels.Server) : PostgresAbstractServer {
```

Comment:
Rather than copy/pasting the properties like this, we can directly use the original object and a `type` instead of an `interface`.

The only complication with `PostgresAbstractServer` is `serverType`. Do we actually need that property or is there already some property that identifies if it's flexible/single? If we _do_ need the property, you can still use the original object, but just add `serverType` on top like this:
```typescript
export type PostgresAbstractServer = (SingleModels.Server | FlexibleModels.Server) & { serverType: PostgresServerType; }
```

```typescript
export function singleAsAbstractServer(server: SingleModels.Server): PostgresAbstractServer {
    return Object.assign(server, { serverType: PostgresServerType.Single });
}
```

Reply:
The models are very different. Doing this is throwing a range of compiler errors

```
Argument of type '(server: Server) => PostgresAbstractServer' is not assignable to parameter of type '((value: Server, index: number, array: Server[]) => PostgresAbstractServer) & ((value: Server, index: number, array: Server[]) => PostgresAbstractServer)'.
  Type '(server: Server) => PostgresAbstractServer' is not assignable to type '(value: Server, index: number, array: Server[]) => PostgresAbstractServer'.
    Types of parameters 'server' and 'value' are incompatible.
      Type 'import("/Users/anthonyshaw/projects/vscode-cosmosdb/node_modules/@azure/arm-postgresql/esm/models/index").Server' is not assignable to type 'import("/Users/anthonyshaw/projects/vscode-cosmosdb/node_modules/@azure/arm-postgresql-flexible/esm/models/index").Server'.
        Types of property 'sku' are incompatible.
          Type 'import("/Users/anthonyshaw/projects/vscode-cosmosdb/node_modules/@azure/arm-postgresql/esm/models/index").Sku | undefined' is not assignable to type 'import("/Users/anthonyshaw/projects/vscode-cosmosdb/node_modules/@azure/arm-postgresql-flexible/esm/models/index").Sku | undefined'.
            Type 'import("/Users/anthonyshaw/projects/vscode-cosmosdb/node_modules/@azure/arm-postgresql/esm/models/index").Sku' is not assignable to type 'import("/Users/anthonyshaw/projects/vscode-cosmosdb/node_modules/@azure/arm-postgresql-flexible/esm/models/index").Sku'.
              Types of property 'tier' are incompatible.
                Type 'SkuTier | undefined' is not assignable to type 'SkuTier'.
                  Type 'undefined' is not assignable to type 'SkuTier'.ts(2345)
```